### ISSUE TITLE
Refactor runtime module for training, validation and inference

### ISSUE THREAD

**Issue by lustiz** [lustiz/zzqnwqgdtdtcqqwyjszpscfz#25]
Follow up on #24 

**Refactor runtime module:**
- [x] training #28
- [x] validation #29 
- [x] testing/inference #30 

**Follow up work:**
- [ ] Make sure timings work.
- [ ] Assertion of NaNs.
- [ ] Progress Description with list of losses.
- [ ] Writing and resuming checkpoints.
- [ ] Hyperparameter Schedule .

**Comment by lustiz** [comment]
**train(args, epoch, start_iteration, data_loader, model, optimizer)**
- model.train()
- tdqm with smoothing
- call model without inference flag
- perform optimization steps
- only obtain loss

**validate(args, epoch, start_iteration, data_loader, model, optimizer)**
- model.eval()
- tdqm without smoothing
- call model without inference flag
- only obtain loss

**inference(args, epoch, data_loader, model)**          //  NO: start_iteration, no optimizer
- model.eval()
- tdqm without smoothing
- call model with inference flag
- obtain loss and outputs

**Comment by lustiz** [comment]
Closed for now
